core/src/main/groovy/com/craigburke/document/core/Image.groovy

```diff
@@ -9,9 +11,45 @@ class Image extends BaseNode {
     ImageType type = ImageType.JPG
     Integer width
     Integer height
-    byte[] data
-    
-    void setType(String value) { 
-        type = Enum.valueOf(ImageType, value.toUpperCase()) 
+    private URL imageUrl
+    private byte[] imageData
+
+    void setType(String value) {
+        type = Enum.valueOf(ImageType, value.toUpperCase())
+    }
+
+    URL getUrl() {
+        imageUrl
+    }
+
+    void setUrl(String value) {
+        setUrl(value != null ? URI.create(value).toURL() : null)
+    }
+
+    void setUrl(URL value) {
```

Comment:
We can simplify this by getting rid of this and `URL getUrl()` I think we only need a simple url String property (and rely on the implicit Groovy getters and setter). I don't really see a situation where we'd need to explicitly set a URL instance. Thoughts?